From f136e7e6804bc02c9a623905d94eac955e2df0a3 Mon Sep 17 00:00:00 2001
From: Eoghan Glynn <eglynn@redhat.com>
Date: Thu, 12 Apr 2012 11:27:18 +0100
Subject: [PATCH] Ensure swift auth URL includes trailing slash

Fixes bug 979745

Image objects in swift were previously leaked post-deletion
due to a silent auth failure caused by the absense of the
trailing forward slash on the swift connection auth URL.

Change-Id: I9c73a2f75a6466e73801ababdd81db77701ccb20
---
 glance/store/swift.py |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/glance/store/swift.py b/glance/store/swift.py
index 2659daa..6db54c4 100644
--- a/glance/store/swift.py
+++ b/glance/store/swift.py
@@ -303,12 +303,14 @@ class Store(glance.store.base.Store):
         """
         snet = self.snet
         auth_version = self.auth_version
+        full_auth_url = (auth_url if not auth_url or auth_url.endswith('/')
+                         else auth_url + '/')
         logger.debug(_("Creating Swift connection with "
-                     "(auth_address=%(auth_url)s, user=%(user)s, "
+                     "(auth_address=%(full_auth_url)s, user=%(user)s, "
                      "snet=%(snet)s, auth_version=%(auth_version)s)") %
                      locals())
         return swift_client.Connection(
-            authurl=auth_url, user=user, key=key, snet=snet,
+            authurl=full_auth_url, user=user, key=key, snet=snet,
             auth_version=auth_version)
 
     def _option_get(self, param):
